demo-circuit/src/naive_threshold_sig_w_key_rotation/mod.rs

```diff
@@ -0,0 +1,1033 @@
+        prev_epoch_number: Option<FieldElement>,
+        prev_end_cumulative_sc_tx_comm_tree_root: Option<FieldElement>,
+        prev_mr_bt: Option<FieldElement>,
+        prev_ft_min_amount: Option<u64>,
+        prev_btr_fee: Option<u64>,
+
+        max_pks: usize,
+        custom_fields: Option<Vec<FieldElement>>,
+        prev_custom_fields: Option<Vec<FieldElement>>,
+
+        genesis_validator_keys_tree_root: Option<FieldElement>,
+
+        prev_cert_data_hash_present: bool,
+    ) -> Self {
+        //Convert needed variables into field elements
+        let fees_field_elements = {
+            let fes = DataAccumulator::init()
+                .update(btr_fee)
+                .unwrap()
+                .update(ft_min_amount)
+                .unwrap()
+                .get_field_elements()
+                .unwrap();
+            assert_eq!(fes.len(), 1);
+            fes[0]
+        };
+
+        //CR: TODO verify this. I use threshold as quality instead of "valid_signatures"
+        //let quality = FieldElement::from(valid_signatures);
+        let quality = threshold;
+
+        //Compute genesis_constant (pks_threshold_hash)
+        let genesis_constant = if prev_cert_data_hash_present {
+            FieldHash::init_constant_length(2, None)
+                .update(genesis_validator_keys_tree_root.unwrap())
+                .update(threshold)
+                .finalize()
+                .unwrap()
+        } else {
+            let mut h = FieldHash::init_constant_length(signing_keys.len(), None);
+            signing_keys.iter().for_each(|pk| {
+                h.update(pk.0.into_affine().x);
+            });
+            let pks_hash = h.finalize().unwrap();
+            FieldHash::init_constant_length(2, None)
+                .update(pks_hash)
+                .update(threshold)
+                .finalize()
+                .unwrap()
+        };
+        //Compute cert_data_hash
+        let cert_data_hash = {
+            // Compute wcert_sysdata_hash
+            let wcert_sysdata_hash = FieldHash::init_constant_length(6, None)
+                .update(sc_id)
+                .update(epoch_number)
+                .update(mr_bt)
+                .update(quality)
+                .update(end_cumulative_sc_tx_comm_tree_root)
+                .update(fees_field_elements)
+                .finalize()
+                .unwrap();
+
+            // Compute custom_fields hash taking into account the presence, or not, of custom_fields
+            let to_hash = if let Some(custom_fields) = custom_fields.clone() {
+                let mut h = FieldHash::init_constant_length(custom_fields.len(), None);
+                custom_fields.into_iter().for_each(|custom_field| {
+                    h.update(custom_field);
+                });
+                let custom_fields_hash = h.finalize().unwrap();
+                vec![custom_fields_hash, wcert_sysdata_hash]
+            } else {
+                vec![wcert_sysdata_hash]
```

Comment:
I think in this particular instance we can safely assume that there will always be at least one custom field